src/main/java/org/wahlzeit_revisited/repository/PhotoRepository.java

```diff
@@ -0,0 +1,143 @@
+import org.wahlzeit_revisited.model.PhotoFactory;
+import org.wahlzeit_revisited.model.PhotoStatus;
+import org.wahlzeit_revisited.model.User;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class PhotoRepository extends AbstractRepository<Photo> {
+
+    @Inject
+    public PhotoFactory factory;
+
+    /*
+     * AbstractRepository contract
+     */
+
+    @Override
+    protected Optional<Photo> doFindById(Long id) throws SQLException {
+        String query = "SELECT * FROM photos WHERE id = ?";
+        PreparedStatement stmt = getReadingStatement(query);
+        stmt.setLong(1, id);
+
+        Photo photo = null;
+        try (ResultSet resultSet = stmt.executeQuery()) {
+            if (resultSet.next()) {
+                photo = parseAsteriskRow(resultSet);
+            }
+        }
+
+        return Optional.ofNullable(photo);
+    }
+
+    @Override
+    protected List<Photo> doFindAll() throws SQLException {
+        String query = "SELECT * FROM photos";
+        PreparedStatement stmt = getReadingStatement(query);
+
+        List<Photo> result = new ArrayList<>();
+        try (ResultSet resultSet = stmt.executeQuery()) {
+            while (resultSet.next()) {
+                Photo photo = parseAsteriskRow(resultSet);
+                result.add(photo);
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    protected Photo doInsert(Photo toInsert) throws SQLException {
+        String query = "INSERT INTO photos (creation_time, owner_id, status, width, height) VALUES(?, ?, ?, ?, ?) RETURNING id";
+        PreparedStatement stmt = getReadingStatement(query);
+        stmt.setLong(1, toInsert.getCreationTime());
+        stmt.setLong(2, toInsert.getOwnerId() == null ? 0 : toInsert.getOwnerId()); // admin is owner
+        stmt.setInt(3, toInsert.getStatus().asInt());
+        stmt.setInt(4, toInsert.getWidth());
+        stmt.setInt(5, toInsert.getHeight());
+
+        // Extract id from RETURNING statement
+        try (ResultSet returningSet = stmt.executeQuery()) {
+            returningSet.next();
+            Long persistedId = returningSet.getLong(1);
+            toInsert.setId(persistedId);
+        }
+
+        return toInsert;
+    }
+
+    @Override
+    protected Photo doUpdate(Photo toUpdate) throws SQLException {
+        String query = "UPDATE photos SET owner_id = ?, status = ?, width = ?, height = ? WHERE id = ?";
+        PreparedStatement stmt = getUpdatingStatement(query);
+        stmt.setLong(1, toUpdate.getOwnerId() == null ? 0 : toUpdate.getOwnerId()); // admin is owner
+        stmt.setInt(2, toUpdate.getStatus().asInt());
+        stmt.setInt(3, toUpdate.getWidth());
+        stmt.setInt(4, toUpdate.getHeight());
+        stmt.setLong(5, toUpdate.getId());
+
+        stmt.execute();
+        return toUpdate;
+    }
+
```

Comment:
So the Serializer pattern is gone? I don't like it that the knowledge of the internal structure of Photo objects is now hard-coded outside the Photo class. The splitting of PhotoManager into PhotoService and PhotoRepository seems to make sense to me, but I would keep Photo internal knowledge out of it. I think you can still have the Photo write its data into a ResultSet object and thereby hide its structure from other classes?

Reply:
So this `rset.updateRow();` in favor for native SQL queries like `"INSERT INTO photos(photo_members)"` and `UPDATE photos SET photo_members` statements?

Just double checking, as in my personal experience it took me quite a while to understand the call-graph and for the `location` related task to persist the `location` class as I used an SQL-JOIN?

Reply:
This would also disallow to use useful database features like `NOT NULL` or `UNIQUE`, and makes Insert and Update operations to race-conditions, as eg an email could get inserted twice

I'm not quite sure, how to consistently implement Database Query-Methodes like `UserRepositoy::findByUsernamePassword(String username, String plainPassword)` - as they would query the database with a `SELECT * FROM email = ?, password = ?` query, where the knowledge of the internal structure is still needed 

But indeed this implementation follows the [Table Data Gateway](https://martinfowler.com/eaaCatalog/rowDataGateway.html) pattern, which is oblivious  not the specified serializer patter
So as the (Scrum) customer, you have the last word here.

Reply:
*[Table Data Gateway](https://martinfowler.com/eaaCatalog/tableDataGateway.html) pattern

Reply:
What if the domain object holds the query strings as well?

Reply:
That would end in a weird coupling between the `Persistent` and the according `Repository`

```java
protected Photo doUpdate(Photo toUpdate) throws SQLException {
    String query = toUpdate.getUpdateQuery();
    PreparedStatement stmt = getUpdatingStatement(query);

    stmt = toUpdate.fillUpdateValues(stmt);
    stmt.execute()
    return toUpdate;
}
```

In `Quarkus` ORM mapping is usually done with [Panache Entities](https://quarkus.io/guides/hibernate-orm-panache), which provides the same methods a repository would have, but as static class methods.
I didn't choose this approach, as I didn't want to change the original `ObjectManager` approach to much and thought all the (manual) ORM would blew up the model classes.

But I will implement it by this week and wait for further feedback then - seems to be the best of all worlds